### PR TITLE
Tweak path linting

### LINKED ISSUE #244: Incorrectly checks xpath and returns an error when publishing
The local path checker is a bit overzealous and mistakes parsing of xml as the user trying to read in data. For example
I am interacting with a web API. I then need to do some parsing of the xml. 

```
 r<-wcPOST()
  start_epoch <- as.numeric(now() - lubridate::days(days_lag))
  
  #Get program number
  xpath <- paste("//data/deployment/argos/last_uplink_date", 
                 "[number() > number(", start_epoch, ")]/../program_number", sep = "")
  program_number <- XML::xpathSApply(XML::xmlParse(content(r,"text")), xpath, XML::xmlValue)
 ```
yields

screen shot 2018-02-12 at 10 25 23 am

This all works great locally and hosted. Note there is no interaction with the local file system. But when I go to publish I get. "Paths should be to files within the project directory". I can force a 'yes' to override the check, but it could be improved.

Thanks.

### LINKED ISSUE #611: "File paths are case-sensitive on deployment server" error when deploying to shinyapps.io
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.7.1.554
    OS Version      : aarch64-apple-darwin20 (64-bit) macOS Monterey 12.5
    R Version       : 4.2.1 (2022-06-23)

### Steps to reproduce the problem

When deploying shiny app to shinyapps.io, an error for review comes up. It appears to arise from the following line:

```
tidyr::separate(value, into = c("description", "value"), sep = ":|=", extra = "merge")
```

Including this line in an R script for an app deployed to shinyapps.io reproduces the issue for me. As best I can tell, its the inclusion of the word "description" in the into argument. removing description appears to fix the issue for me.

### Describe the problem in detail
 
image

image

### Describe the behavior you expected

As far as I can tell, this is not an error. Apologies if I'm mistaken or have missed a previous report of this issue.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### PR DESCRIPTION
**Pull request by hadley**
* Remove "bad relative path linker". This used a very simple heuristic that had many possible false positives. Fixes #244.

* Only check case of potential paths that contain a `/`. This reduces the number of false positives. Fixes #611.

### CODE REVIEW

File: R/linters.R [hunk 1]
```diff
@@ -169,6 +149,8 @@ addLinter("filepath.capitalization", linter(
       lapply(regex, function(x) {
 
         which(
+          # Only examine paths containing a slash to reduce false positives
```
Comment (aronatkins) [inline R/linters.R:152]: The side-effect, unfortunately, is that a file named `MyData.csv` but referenced as `mydata.csv` will go uncaught. So sad.

File: tests/testthat/shinyapp-with-absolute-paths/server.R [hunk 2]
```diff
@@ -28,8 +28,8 @@ shinyServer(function(input, output) {
     hist(x, breaks = bins, col = "darkgray", border = "white")
 
     ## read a csv file
-    file <- read.csv("college.txt") ## bad
-    file <- read.csv("College.txt") ## okay
+    file <- read.csv("data/college.txt") ## bad
```
Comment (aronatkins) [inline tests/testthat/shinyapp-with-absolute-paths/server.R:31]: Maybe also add a test demonstrating that we do not detect problems with a current-directory `college.txt`? Very weak nit.

**Review by aronatkins — APPROVED** [review]

File: R/linters.R (hunk 1 above)
Comment (hadley) [reply R/linters.R:152]: Yeah, it's a bummer. I have racked my brains to come up with another technique to use, but I have failed so far.

Comment (aronatkins) [reply R/linters.R:152]: An option that comes to mind for me is to adjust some of the file-reading utilities; if they experience a file-not-found error, they could look for a case-insensitive same-name file and add that context to the error message.

> File `mydata.csv` is not found, but there is a file named `MyData.csv`.

This feels excessive in most cases, though.. and we wouldn't be able to adjust all such functions.

**Pull request merged**